### PR TITLE
Spatial: Test NULL more

### PR DESCRIPTION
**Pull request by bricelam**
Also adds SQL Server coverage for constants with non-default SRIDs

### CODE REVIEW

**Comment by bricelam** [comment]
Interestingly, SpatiaLite returns true for any boolean operation on a null making it different from both C# and Transact-SQL.

**Comment by bricelam** [comment]
As discussed, I'm going to update the boolean translations for SpatiaLite.

**Comment by bricelam** [comment]
Updated. Good to review.

File: src/EFCore.Relational/Query/Sql/DefaultQuerySqlGenerator.cs [hunk 1]
```diff
@@ -2140,13 +2140,56 @@ protected override Expression VisitExtension(Expression extensionExpression)
...
+                {
+                    case SelectExpression selectExpression:
+                        // We skip visiting SelectExpression here because it will be processed by outer visitor when
+                        // generating SQL
+                        newExpression = selectExpression;
+                        break;
+
+                    case CaseExpression caseExpression:
+                        _isSearchCondition = false;
+                        var newOperand = caseExpression.Operand != null
```
Comment (smitpatel) [inline src/EFCore.Relational/Query/Sql/DefaultQuerySqlGenerator.cs]: You shouldn't need a null check here. Visit(null) returns null;

**Pull request merged**